app/src/main/java/net/kwatts/powtools/util/BluetoothUtilImpl.java

```diff
@@ -166,15 +167,17 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status){
         @Override
         public void onCharacteristicRead(BluetoothGatt gatt, BluetoothGattCharacteristic c, int status) {
             String characteristic_uuid = c.getUuid().toString();
-            Log.i(TAG, "BluetoothGattCallback.onCharacteristicRead: CharacteristicUuid=" + characteristic_uuid + "status=" + status);
+            Timber.i( "BluetoothGattCallback.onCharacteristicRead: CharacteristicUuid=" + characteristic_uuid + "status=" + status);
             characteristicReadQueue.remove();
 
 
             //XXX until we figure out what's going on
             if (characteristic_uuid.equals(OWDevice.OnewheelCharacteristicBatteryRemaining)) {
                 mainActivity.updateBatteryRemaining(c.getIntValue(BluetoothGattCharacteristic.FORMAT_UINT8, 1));
-            }  else if (characteristic_uuid.equals(OWDevice.OnewheelCharacteristicRidingMode)) {
-                 Log.d(TAG, "Got ride mode from the main UI thread:" + c.getIntValue(BluetoothGattCharacteristic.FORMAT_UINT8, 1));
+            } else if (c.getUuid().toString().equals(OWDevice.OnewheelCharacteristicSpeedRpm)) {
+                mainActivity.updateCurrentSpeed(c.getIntValue(BluetoothGattCharacteristic.FORMAT_UINT8, 1));
```

Comment:
instead of doing this, it seems cleaner to add a listener on the observable field you added in this same PR. The less reliance on a god activity the better.